### ISSUE TITLE
ColorScheme onInverseSurface does not follow M3 specification.

### ISSUE THREAD

**Issue by CITguy** [flutter/flutter#159501]
### Steps to reproduce

`ColorScheme` incorrectly defines [`onInverseSurface`](https://api.flutter.dev/flutter/material/ColorScheme/onInverseSurface.html) as the property to configure the "Inverse On Surface" color role.  This is confusing when comparing Flutter configurations against M3 specifications.

### Expected results

`inverseOnSurface` should be used consistently within Flutter assets to refer to the "Inverse On Surface" color role.

### Actual results

**Compliant**

* `MaterialDynamicColors` uses [`inverseOnSurface`](https://api.flutter.dev/flutter/package-material_color_utilities_dynamiccolor_material_dynamic_colors/MaterialDynamicColors/inverseOnSurface.html)
* `DynamicScheme` uses [`inverseOnSurface`](https://api.flutter.dev/flutter/package-material_color_utilities_dynamiccolor_dynamic_scheme/DynamicScheme/inverseOnSurface.html)
* `Scheme` (deprecated) uses [`inverseOnSurface`](https://api.flutter.dev/flutter/package-material_color_utilities_scheme_scheme/Scheme/inverseOnSurface.html)

**Non-compliant**

* `ColorScheme` uses [`onInverseSurface`](https://api.flutter.dev/flutter/material/ColorScheme/onInverseSurface.html)

### Code sample

n/a

### Screenshots or Video

[M3 Color roles - Inverse colors](https://m3.material.io/styles/color/roles#7fc6b47e-db22-4e98-8359-7649a099e4a1)
![PixelSnap 2024-11-26 at 12 37 41@2x](https://github.com/user-attachments/assets/4c3be23e-92ff-4c91-962a-7269ed59363a)

### Logs

n/a

### Flutter Doctor output

n/a

**Comment by TahaTesser** [comment]
cc: @rydmike

**Comment by rydmike** [comment]
The **Flutter SDK Material-3 implementation** contains many APIs where the naming differs a bit from the equivalent in the **Material-3 Design spec** (e.g. Chips and its variants). 

While this is unfortunate, updating all of them to match M3 spec naming exactly, while certainly doable, would also be a bit disruptive and require introducing deprecations and eventually breaking changes.

The **contrasting on** color for **inverse surface** has for some reason in **Flutter** always been called `onInverseSurface` instead of `inverseOnSurface`, like its token is called in M3 spec, if we go by `md.sys.color.inverse-on-surface`. 

The used default color tone and thus the color value for the `onInverseSurface` color in **Flutter**, correctly uses neutral palette tone 95 in light mode, and tone 20 in dark mode. So from that point of view there is no issue with the color value not matching the spec. It is only a naming issue. It is a **correct observation that is not 100% aligned** with the spec naming. **Question is**, if it is worth amending it? That is something the Flutter team needs to make a call on.

I suspect the Flutter naming came from that **all** other contrasting colors in the **ColorScheme** have the **on** qualifier prepended to the color it is a contrasting color for. This color got its name when it was made, based on this naming pattern, without looking closely at what was actually used in the spec.  I noticed this naming inconsistency too when the color appeared some years ago, when implementing support for it in my more flexible seed generation algorithm.

In a way, this Flutter name could be argued to be a **more logical and consistent naming** for the **on** color for the `inverseSurface` color, as it matches naming used for all other contrasting "**on**" colors in the color scheme. You could just as well argue that the naming used in the Material-3 spec is a bit inconsistent for this particular on color.

> [!NOTE]
> We can also notice that `onSurfaceVariant` is no longer a direct "on" color for any currently existing color name, as there no longer is a color called `surfaceVariant` after it was deprecated. It should rather be seen as less distinct **on** contrast color for any of the existing none inverted surface colors, to be used when less emphasis is desired. Still with that, saying that the "on" moniker is always prepended to the name of the color it is the contrasting color for, no longer hold for this color either, but at least it is a variant contrasting coloring for all the surface colors.

### Other things worth mentioning about "on" colors and their naming

In the issue that lists "APIs we wish were different", **all** the "**on**" color names are mentioned as being unfortunate names, as they make the colors sound like **event handlers**, a very valid point imo too. (Sorry I don't have the link to this issue handy, but I just happened to re-read it a week ago or so and I recalled this "onColor" naming topic mentioned in it.)

No idea what would be considered a better alternative, maybe instead e.g. `primary` and `primaryContrast` or something similar. Going this way would also be a big deviation from the M3 spec naming.

**Comment by TahaTesser** [comment]
Thanks so much for your input @rydmike!

**Comment by TahaTesser** [comment]
I agree with Mike. Updating the naming convention will cause disruptions and may not be worth it.
